### PR TITLE
stratisd: init at 3.2.2

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> to test:
> 
> ```nix
> {
>   systemd.packages = [ pkgs.stratisd ];
>   services.dbus.packages = [ pkgs.stratisd ];
>   services.udev.packages = [ pkgs.stratisd ];
>   systemd.services.stratisd.wantedBy = [ "sysinit.target" ];
> }
> ```

We should move that into a nixos module, right?

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    rm -r "$out/lib/dracut"
+    rm -r "$out/lib/systemd/system-generators"
+  '';
+
+  meta = with lib; {
+    description = "Easy to use local storage management for Linux";
+    homepage = "https://stratis-storage.github.io";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ nickcao ];
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:113]: aarch64 does not build, see ofborg

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    udev
+  ];
+
+  makeFlags = [ "PREFIX=${placeholder "out"}" ];
+  buildFlags = [ "release" "release-min" "docs/stratisd.8" ];
+
+  doCheck = true;
+  checkTarget = "test";
+
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:88]: postPatch normally goes straight after src.

**Comment by NickCao** [comment]
> We should move that into a nixos module, right?

I'd like to take care of that and add some tests later.

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+, jose
+, jq
+, curl
+, tpm2-tools
+, coreutils
+, clevisSupport ? false
+}:
+
+let
+  runtimeBinaries = [
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:25]: We should probably inline this.

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise looks solid

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  src = fetchFromGitHub {
+    owner = "stratis-storage";
+    repo = finalAttrs.pname;
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-u21mTgg36RzccJe7RQiR/MF5lFs6vmjXOkaGaHNdB+4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:36]:
```suggestion
    inherit src;
```

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "stratisd";
+  version = "3.2.1";
+
+  src = fetchFromGitHub {
+    owner = "stratis-storage";
+    repo = finalAttrs.pname;
+    rev = "v${finalAttrs.version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:31]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/filesystems/stratisd/default.nix [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+, clevis
+, jose
+, jq
+, curl
+, tpm2-tools
+, coreutils
+, clevisSupport ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/stratisd/default.nix:24]:
```suggestion
stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — COMMENTED** [review]
I just noticed that this will not work since both variables are fixed output derivations

**Comment by NickCao** [comment]
> I just noticed that this will not work since both variables are fixed output derivations

There are works on making these fetchers overriable: https://github.com/NixOS/nixpkgs/pull/158018

**Comment by SuperSandro2000** [comment]
Can we bush that back until it is figured out how to do that? Right now it is really vague and nothing concrete. Maybe this gets so big that we get a treewide change and convert everything? We don't know yet and I would like to avoid compatibility code for something that might come eventually.

**Pull request merged**